src/effects/effectsmanager.cpp

```diff
@@ -399,15 +399,24 @@ void EffectsManager::setup() {
 //     m_pEffectChainManager->loadEffectChains();
 // }
 
+inline void EffectsManager::refreshChainSlot(EffectChainSlotPointer pChainSlot) {
```

Comment:
Why did you move this from EffectChainSlot?

Reply:
Earlier the flow was DlgPrefLV2 -> EffectsManager::refreshAllChainSlots -> EffectChainSlot::refreshAllEffects -> EffectsManager::loadEffect -> EffectChainSlot::loadEffect -> EffectSlot::loadEffect. And the EffectsManager::loadEffect function required an Shared pointer of the EffectChainSlot. Moving this functionality to EffectsManager resolves the above issues and simplifies the flow.

Reply:
Okay, thank you for explaining. I see what the problem was, but I think we can come up with a better solution than this. EffectChainSlot::refreshAllEffects should not have needed to make a call to EffectsManager which required a pointer back to the EffectChainSlot which made the call. I think we can make an EffectSlot::refresh function if EffectSlot::loadEffect stores the EffectInstantiatorPointer it is passed as a member variable.